# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1424782382 -3600
#      Tue Feb 24 13:53:02 2015 +0100
# Node ID 29692bc97899e0124f8de47c8c05b41b879640bf
# Parent  a09471d938a51df3e97ae5a4c2ba11832c6c47fd
fix font in windows terminal when scaling is 150% or more (bug #41938)

* QWinTerminalImpl.cpp (updateConsoelSize): get font metrics from console
  widget and not from the font object

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -980,17 +980,17 @@ void QConsolePrivate::log (const char* f
   Q_UNUSED (fmt);
 #endif
 }
 
 //////////////////////////////////////////////////////////////////////////////
 
 void QConsolePrivate::updateConsoleSize (bool sync, bool allow_smaller_width)
 {
-  QFontMetrics fm (m_font);
+  QFontMetrics fm = m_consoleView->fontMetrics ();
   QSize winSize = m_consoleView->size ();
 
   m_charSize.rwidth () = fm.averageCharWidth ();
   m_charSize.rheight () = fm.lineSpacing ();
 
   m_consoleRect.setWidth (winSize.width () / fm.averageCharWidth ());
   m_consoleRect.setHeight (winSize.height () / fm.lineSpacing ());
 
